question-servers/freeform.js

```diff
@@ -199,7 +217,7 @@ module.exports = {
             const courseErr = new Error('Invalid state before ' + phase + ': ' + checkErr);
             courseErr.fatal = true;
             courseErrs.push(courseErr);
-            return callback(null, courseErrs, data, '', fileData);
+            return callback(null, courseErrs, isGetDependencies ? elementDependencies : data, '', fileData);
```

Comment:
We have been adding new arguments for each type of return data (so we have `data, html, fileData`). This would seem to suggest adding `, dependencies` to the end of this, but it seems increasingly horrible. Maybe we should collapse all of them to `retVal` or something? This isn't really necessary, of course.